src/db/pool.cr

```diff
@@ -91,10 +100,19 @@ module DB
         resource
       end
 
-      res.before_checkout
+      if res.responds_to?(:before_checkout)
+        res.before_checkout
+      end
       res
     end
 
+    def checkout(&block : T ->)
```

Comment:
I'm not sure. But the new `Pool#checkout(&)` might not play nice with DB connections. It's not documented, but the `auto_release` property handles the logic of how the resource is released after the operation is completed. This requires some knowledge of the resource though. Maybe we need to formalise that a bit better.

Reply:
Interesting. I haven't looked into how that works but it sounds like it might fall under `DB`-specific extensions that could be made if this class were extracted to its own shard, which `DB::Pool` could then inherit from.